### PR TITLE
[Docs] Add sorting and source filtering section to client docs

### PR DESCRIPTION
**Pull request by cbuescher**
Add a short section on sorting and fetching `_source`.

### CODE REVIEW

File: docs/java-rest/high-level/apis/search.asciidoc [hunk 1]
```diff
@@ -82,6 +82,34 @@ After this, the `SearchSourceBuilder` only needs to be added to the
...
+===== Source filtering
+
+By default, search requests return the contents of the document `_source` but like in the Rest API you can overwrite this behaviour. For example, you can turn off `_source` retrieval completely:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-source-filtering-off]
+--------------------------------------------------
+
+The method also accepts and array of one or more wildcard patterns to control which fields get included or excluded in a more fine grained way:
```
Comment (javanna) [inline docs/java-rest/high-level/apis/search.asciidoc]: s/and/an

File: docs/java-rest/high-level/apis/search.asciidoc [hunk 2]
```diff
@@ -82,6 +82,34 @@ After this, the `SearchSourceBuilder` only needs to be added to the
...
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-source-filtering-includes]
+--------------------------------------------------
```
Comment (javanna) [inline docs/java-rest/high-level/apis/search.asciidoc]: you can probably reuse the examples that we already have in the get API here.

**Review by javanna — APPROVED** [review]
left a couple of comments, LGTM otherwise

File: docs/java-rest/high-level/apis/search.asciidoc (hunk 2 above)
Comment (cbuescher) [inline docs/java-rest/high-level/apis/search.asciidoc]: I looked at the example in CrudDocumentationIT but it sets the FetchSourceContext on an UpdateRequest, and I'd like to use the simpler version using `SearchSourceBuilder#fetchSource(@Nullable String[] includes, @Nullable String[] excludes)` here, so I'm leaving this as is.

Comment (javanna) [inline docs/java-rest/high-level/apis/search.asciidoc]: good point that's fine

**Pull request merged**